product-service/serverless.ts

```diff
@@ -0,0 +1,57 @@
+import type { Serverless } from 'serverless/aws';
+
+const serverlessConfiguration: Serverless = {
+  service: {
+    name: 'product-service',
+    // app and org for use with dashboard.serverless.com
+    // app: your-app-name,
+    // org: your-org-name,
+  },
+  frameworkVersion: '2',
+  custom: {
+    webpack: {
+      webpackConfig: './webpack.config.js',
+      includeModules: true
+    }
+  },
+  // Add the serverless-webpack plugin
+  plugins: ['serverless-webpack'],
+  provider: {
+    name: 'aws',
+    runtime: 'nodejs12.x',
+    apiGateway: {
+      minimumCompressionSize: 1024,
+    },
+    environment: {
+      AWS_NODEJS_CONNECTION_REUSE_ENABLED: '1',
+    },
+  },
+  functions: {
+    getAllProducts: {
+      handler: 'handler.getAllProducts',
+      events: [
+        {
+          http: {
+            method: 'get',
+            path: 'products',
+            cors: true,
+          }
+        }
+      ]
+    },
+    getProductById: {
+      handler: 'handler.getProductById',
+      events: [
+        {
+          http: {
+            method: 'get',
+            path: 'products/{productId}',
+            cors: true,
```

Comment:
or  
provider 
  httpApi:
    cors: true
And use httpApi instead of http

Reply:
Moved `cors: true` to `httpApi` and reused `httpApi` with function